lifecycle-operator/webhooks/pod_mutator/handlers/appcreationrequest_handler.go

```diff
@@ -0,0 +1,97 @@
+	EventSender controllercommon.IEvent
+}
+
+func (a *AppCreationRequestHandler) Handle(ctx context.Context, pod *corev1.Pod, namespace string) error {
+
+	ctx, span := a.Tracer.Start(ctx, "create_app", trace.WithSpanKind(trace.SpanKindProducer))
+	defer span.End()
+
+	newAppCreationRequest := generateAppCreationRequest(ctx, pod, namespace)
+	newAppCreationRequest.SetSpanAttributes(span)
+
+	a.Log.Info("Searching for AppCreationRequest", "appCreationRequest", newAppCreationRequest.Name, "namespace:", newAppCreationRequest.Namespace)
+
+	appCreationRequest := &klcv1alpha3.KeptnAppCreationRequest{}
+	err := a.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: newAppCreationRequest.Name}, appCreationRequest)
+	if errors.IsNotFound(err) {
+		return a.createApp(ctx, newAppCreationRequest, span)
+	}
+
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+		return fmt.Errorf("could not fetch AppCreationRequest"+": %+v", err)
+	}
+	a.Log.Info("Found AppCreationRequest", "appCreationRequest", newAppCreationRequest.Name, "namespace:", newAppCreationRequest.Namespace)
+	return nil
+}
+
+func (a *AppCreationRequestHandler) createApp(ctx context.Context, newAppCreationRequest *klcv1alpha3.KeptnAppCreationRequest, span trace.Span) error {
+	a.Log.Info("Creating app creation request", "appCreationRequest", newAppCreationRequest.Name, "namespace:", newAppCreationRequest.Namespace)
+
+	err := a.Client.Create(ctx, newAppCreationRequest)
+	if err != nil {
+		a.Log.Error(err, "Could not create AppCreationRequest")
+		a.EventSender.Emit(apicommon.PhaseCreateAppCreationRequest, "Warning", newAppCreationRequest, apicommon.PhaseStateFailed, "could not create KeptnAppCreationRequest", newAppCreationRequest.Spec.AppName)
+		span.SetStatus(codes.Error, err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func generateAppCreationRequest(ctx context.Context, pod *corev1.Pod, namespace string) *klcv1alpha3.KeptnAppCreationRequest {
+
+	// create TraceContext
+	// follow up with a Keptn propagator that JSON-encoded the OTel map into our own key
+	traceContextCarrier := propagation.MapCarrier{}
+	otel.GetTextMapPropagator().Inject(ctx, traceContextCarrier)
+
+	kacr := &klcv1alpha3.KeptnAppCreationRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   namespace,
+			Annotations: traceContextCarrier,
+		},
+	}
+
+	if !isAppAnnotationPresent(&pod.ObjectMeta) {
+		initEmptyAnnotations(&pod.ObjectMeta)
+		// at this point if the pod does not have an app annotation it means we create the app
+		// and it will have a single workload
+		pod.ObjectMeta.Annotations[apicommon.AppAnnotation] = pod.ObjectMeta.Annotations[apicommon.WorkloadAnnotation]
+		// so we can mark the app request as single service type
+		kacr.Annotations[apicommon.AppTypeAnnotation] = string(apicommon.AppTypeSingleService)
+	}
+
+	appName := getAppName(&pod.ObjectMeta)
+	kacr.ObjectMeta.Name = appName
+	kacr.Spec = klcv1alpha3.KeptnAppCreationRequestSpec{
+		AppName: appName,
+	}
```

Comment:
Let's add this info directly to the object when creating it -> on line 74

Reply:
we cannot since we need for line 87 first